framework/src/modules/interoperability/utils.ts

```diff
@@ -175,3 +211,327 @@ export const swapReceivingAndSendingChainIDs = (ccm: CCMsg) => ({
 	receivingChainID: ccm.sendingChainID,
 	sendingChainID: ccm.receivingChainID,
 });
+export const rawStateStoreKey = (storePrefix: number) => {
+	const moduleIDBuffer = Buffer.alloc(4);
+	moduleIDBuffer.writeInt32BE(MODULE_ID_INTEROPERABILITY, 0);
+	const storePrefixBuffer = Buffer.alloc(2);
+	storePrefixBuffer.writeUInt16BE(storePrefix, 0);
+
+	return Buffer.concat([DB_KEY_STATE_STORE, moduleIDBuffer, storePrefixBuffer]);
+};
+
+export const isInboxUpdateEmpty = (inboxUpdate: InboxUpdate) =>
+	inboxUpdate.crossChainMessages.length === 0 &&
+	inboxUpdate.messageWitness.siblingHashes.length === 0 &&
+	inboxUpdate.messageWitness.partnerChainOutboxSize === BigInt(0) &&
+	inboxUpdate.outboxRootWitness.siblingHashes.length === 0 &&
+	inboxUpdate.outboxRootWitness.bitmap.length === 0;
+
+export const isCertificateEmpty = (decodedCertificate: Certificate) =>
+	decodedCertificate.blockID.equals(EMPTY_BYTES) ||
+	decodedCertificate.stateRoot.equals(EMPTY_BYTES) ||
+	decodedCertificate.validatorsHash.equals(EMPTY_BYTES) ||
+	decodedCertificate.timestamp === 0;
+
+export const isMessageWitnessEmpty = (messageWitness: MsgWitness) =>
+	messageWitness.partnerChainOutboxSize === BigInt(0) && messageWitness.siblingHashes.length === 0;
+
+export const checkLivenessRequirementFirstCCU = (
+	partnerChainAccount: ChainAccount,
+	txParams: CrossChainUpdateTransactionParams,
+): VerificationResult => {
+	if (partnerChainAccount.status === CHAIN_REGISTERED && txParams.certificate.equals(EMPTY_BYTES)) {
+		return {
+			status: VerifyStatus.FAIL,
+			error: new Error(
+				`Sending partner chain ${txParams.sendingChainID} is in registered status so certificate cannot be empty.`,
+			),
+		};
+	}
+
+	return {
+		status: VerifyStatus.OK,
+	};
+};
+
+export const checkCertificateValidity = (
+	partnerChainAccount: ChainAccount,
+	encodedCertificate: Buffer,
+): VerificationResult => {
+	if (encodedCertificate.equals(EMPTY_BYTES)) {
```

Comment:
So the first condition checks whether a certificate is non-empty, empty buffer is like undefined, in that case we don't check the validity of certificate. whereas at L269  if certificate has non-empty buffer and we decode it then it should have all the required values in certificate

Reply:
I mean both are cases for empty certificate. For one the verify status `OK` doesn't make sense when the other is `FAIL`. I think it should the return the same status(with a diff error msg) in both the cases

Reply:
According to https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#certificate-and-validators-update-validity,
```
If params contains a non-empty certificate, it is valid if:
```

from my understanding, it says if a certificate is present then we need to check its validity and missing values, else we skip this check and hence it will be status OK

Reply:
mm yea the LIP is not super clear about it but I think we need to clarify it since missing all values and empty buffer is the same case (i.e. empty certificate) in my understanding.

Reply:
Checking with research team, will be tackled in https://github.com/LiskHQ/lisk-sdk/issues/7185